### PR TITLE
Enable time parameters for TestCase

### PR DESCRIPTION
**Pull request by Luchsor**
Hi,
warningTime and criticalTime parameters in the TestCase constructor don't seem to have any effect regarding the TestExecutionContext. This should load them into the TestExecutionContext so the TestCase can be initiated as in sakuli1. 

```
(async () => {
    const testCase = new TestCase("Checking Sakuli.io", 10, 15);
...
```

Best regards

### CODE REVIEW

File: packages/sakuli-legacy/src/context/common/test-case/test-case.class.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export function createTestCaseClass(ctx: TestExecutionContext, project: Project,
             public _imagePaths: string[] = []
         ) {
             ctx.logger.info(`Start Testcase ${caseId}`);
-            ctx.startTestCase({id: caseId});
+            ctx.startTestCase({id: caseId, warningTime: warningTime, criticalTime: criticalTime});
```
Comment (s1hofmann) [inline packages/sakuli-legacy/src/context/common/test-case/test-case.class.ts]: With ES2015 it's possible to use a shorthand notation to set object properties from variables with matching names, so we could shorten this call.
See [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions) for reference.

**Comment by Luchsor** [comment]
Im not sure why the ci isnt building - the only error i can detect is that the sonar-scanner can't be found but shouldnt that be a problem with any build?! https://travis-ci.com/sakuli/sakuli/jobs/224284850#L1678 (if there's another problem i'd appreciate a hint :D)

In terms of the `test-case.class.spec.ts` update: do u need any further unittest or is passing dummy parameters to all the instances enough?

Thanks for ur effort! 👍

**Comment by tnobody** [comment]
Hi @Luchsor,

it seems to be an problem with travis and sonar: https://github.com/travis-ci/travis-ci/issues/10062

We will discuss a possible solution/workaround and provide it as soon as possible.

Any way thank for your efforts :)

Br Tim

**Comment by s1hofmann** [comment]
Hi @Luchsor,

the `TestCase` constructor comes with default parameters for `warningTime` and `criticalTime`.
The thing we actually want to verify is that the `TestExecutionContext` is updated properly, given these parameters.

So, given some thresholds and an id,

```
const testCaseId = "testCaseId";
const warningTime = 10;
const criticalTime = 15;
```

when creating a new `TestCase` instance

```
const tc = new TestCase(testCaseId, warningTime, criticalTime);
```

we would expect the `TestExecutionContext` to start a new `TestCase`

```
expect(testExecutionContext.startTestCase).toBeCalledWith({
                id: testCaseId,
                warningTime,
                criticalTime
            });
```

This could be placed in a separate block in the `TestCase` spec.

Let us know if you have any further questions.

Best regards

Simon

**Pull request merged**